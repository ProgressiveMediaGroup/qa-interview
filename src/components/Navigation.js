// Node_Modules
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

const navitems = [
  {
    url: '/',
    title: 'Home'
  },
  {
    url: '/automated/test-case-1',
    title: 'Automated QA: Test Case 1'
  },
  {
    url: '/automated/test-case-2',
    title: 'Automated QA: Test Case 2'
  },
  {
    url: '/manual/test-case-1',
    title: 'Manual QA: Test Case 1'
  }
];

class Navigation extends Component {

  constructor() {
    super();
    this.renderNavItem = this.renderNavItem.bind(this);
  }

  renderNavItem (item,index) {
    if (this.props.url === item.url) {
      return <li><Link to={item.url} className="selected">{item.title}</Link></li>
    }

    return <li><Link to={item.url}>{item.title}</Link></li>
  }

  renderNavItems () {
    return navitems.map(this.renderNavItem)
  }

  render () {
    return (
      <ul>
        {this.renderNavItems()}
      </ul>
    )
  }
}

export default Navigation
