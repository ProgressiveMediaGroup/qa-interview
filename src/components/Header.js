// Node_Modules
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

// Assets
import Logo from '../assets/pt-logo.png'

// Styles
import '../styles/header.css'

// Display Components
import Navigation from './Navigation'

class Header extends Component {
  render () {
    return (
      <header>
        <div className='top'>
          <div className='container'>
            <Link to='/'>
              <img src={Logo} alt='Progressive Tech' className='logo-img' />
            </Link>
          </div>
        </div>
        <div className='menu'>
          <div className='container'>
            <Navigation />
          </div>
        </div>
        <div className='breadcrumbs'>
          <div className='container'>
            <Link to='/'>/ Home /</Link>
          </div>
        </div>
      </header>
    )
  }
}

export default Header
