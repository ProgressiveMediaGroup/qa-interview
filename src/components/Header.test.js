/* global it, describe, expect */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter, Link } from 'react-router-dom'
import Header from './Header'
import Navigation from './Navigation'

it('renders without crashing', () => {
  shallow(<Header />)
})

it('should match snapshot', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
})

describe('Has correct root block structure', function () {
  it('header tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('header').length).toBe(1)
  })
  it('top tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('header').find('div.top').length).toBe(1)
  })
  it('menu tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('header').find('div.menu').length).toBe(1)
  })
  it('breadcrumbs tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('header').find('div.breadcrumbs').length).toBe(1)
  })
})

describe('Has correct top block structure', function () {
  it('container tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').find('div.container').length).toBe(1)
  })
  it('link is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').find('div.container').children().at(0).type()).toBe(Link)
  })
  it('link goes to root', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').find('div.container').children().at(0).props().to).toBe('/')
  })
  it('Image is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').children().at(0).find('img').length).toBe(1)
  })
  it('Image is correct', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').children().at(0).find('img').props().src).toBe('pt-logo.png')
  })
  it('Image alt is correct', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').children().at(0).find('img').props().alt).toBe('Progressive Tech')
  })
  it('Image class is correct', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.top').children().at(0).find('img').hasClass('logo-img')).toBe(true)
  })
})

describe('Has correct menu block structure', function () {
  it('container tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.menu').find('div.container').length).toBe(1)
  })
  it('Only one link is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.menu').find('div.container').children().length).toBe(1)
  })
  it('Navigation is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.menu').find('div.container').children().at(0).type()).toBe(Navigation)
  })
})

describe('Has correct breadcrumb block structure', function () {
  it('container tag is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.breadcrumbs').find('div.container').length).toBe(1)
  })
  it('Only one link is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.breadcrumbs').find('div.container').children().length).toBe(1)
  })
  it('link is displayed', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.breadcrumbs').find('div.container').children().at(0).type()).toBe(Link)
  })
  it('link goes to root', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.breadcrumbs').find('div.container').children().at(0).props().to).toBe('/')
  })
  it('link text is correct', () => {
    const wrapper = mount(<MemoryRouter>
      <Header />
    </MemoryRouter>)
    expect(wrapper.find('div.breadcrumbs').find('div.container').children().at(0).text()).toBe('/ Home /')
  })
})
