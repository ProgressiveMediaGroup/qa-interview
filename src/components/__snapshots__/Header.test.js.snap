// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="top"
>
          <div
                    className="container"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              <img
                                        alt="Progressive Tech"
                                        className="logo-img"
                                        src="pt-logo.png"
                              />
                    </Link>
          </div>
</div>,
        <div
          className="menu"
>
          <div
                    className="container"
          >
                    <Navigation />
          </div>
</div>,
        <div
          className="breadcrumbs"
>
          <div
                    className="container"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              / Home /
                    </Link>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <img
                                    alt="Progressive Tech"
                                    className="logo-img"
                                    src="pt-logo.png"
                        />
            </Link>
</div>,
          "className": "top",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              <img
                            alt="Progressive Tech"
                            className="logo-img"
                            src="pt-logo.png"
              />
</Link>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Progressive Tech"
                className="logo-img"
                src="pt-logo.png"
/>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Progressive Tech",
                "className": "logo-img",
                "src": "pt-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <Navigation />
</div>,
          "className": "menu",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Navigation />,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <Link
                        replace={false}
                        to="/"
            >
                        / Home /
            </Link>
</div>,
          "className": "breadcrumbs",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              / Home /
</Link>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "/ Home /",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "/ Home /",
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top"
>
            <div
                        className="container"
            >
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    <img
                                                alt="Progressive Tech"
                                                className="logo-img"
                                                src="pt-logo.png"
                                    />
                        </Link>
            </div>
</div>,
          <div
            className="menu"
>
            <div
                        className="container"
            >
                        <Navigation />
            </div>
</div>,
          <div
            className="breadcrumbs"
>
            <div
                        className="container"
            >
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    / Home /
                        </Link>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <Link
                            replace={false}
                            to="/"
              >
                            <img
                                          alt="Progressive Tech"
                                          className="logo-img"
                                          src="pt-logo.png"
                            />
              </Link>
</div>,
            "className": "top",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                <img
                                alt="Progressive Tech"
                                className="logo-img"
                                src="pt-logo.png"
                />
</Link>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Progressive Tech"
                  className="logo-img"
                  src="pt-logo.png"
/>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Progressive Tech",
                  "className": "logo-img",
                  "src": "pt-logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <Navigation />
</div>,
            "className": "menu",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Navigation />,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <Link
                            replace={false}
                            to="/"
              >
                            / Home /
              </Link>
</div>,
            "className": "breadcrumbs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                / Home /
</Link>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "/ Home /",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "/ Home /",
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
